docs: add calcite-components-react deprecation notice (#13660)
Esri/calcite-components: packages/components-react/README.md

This package provides React wrappers for [Calcite components](https://developers.arcgis.com/calcite-design-system/components/). Refer to the [React example](https://github.com/Esri/calcite-design-system/tree/dev/examples/components/react) for a minimal application using this package.

> ## Deprecation Notice

Author: I tried to add an icon here so it stood out more, but it broke the anchor tag linking functionality. Not sure if there's another way we want to make this stand out more.

Reviewer: A [markdown Alert](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) might be a good use here underneath the heading.

Author: I had tried these earlier as well, but was concerned because it didn't show up correctly in the preview. 

<img width="716" height="80" alt="image" src="https://github.com/user-attachments/assets/e7468af2-0c1f-4908-a66f-fb50be85492d" />

Reviewer: Oh odd, I guess VSCode doesn't support GitHub flavored markdown (GFM)? I always copy the content into a GitHub editor to test. Once it's pushed to a branch you can preview the file in GitHub's UI.

Author: TIL I could preview markdown files in Github. 

It looks like it's working. 🎉 Interestingly, I had tried to Google if it would work and asked Github Copilot. Copilot told me `That GitHub documentation page is misleading...`. 😂

> Calcite Components React is deprecated in v5.0.0 and will be fully removed in v6.0.0 (targeted for February 2027). The `@esri/calcite-components-react` package was originally developed for use with React 18 where wrappers were necessary to use custom elements. With React 19's custom element support, these wrappers are no longer needed.

Reviewer: I wonder if we need to callout the original need in detail here, and unsure if we need to mention the package name again. Take this suggestion with a grain of salt 😆

```suggestion
Calcite Components React is deprecated in v5.0.0 and will be fully removed in v6.0.0 (targeted for February 2027). Originally developed for earlier versions of React where [wrappers were necessary to use custom elements](#why-not-just-use-the-web-components-directly), with React 19's custom element support, these wrappers are no longer needed.
```

This package provides React wrappers for [Calcite components](https://developers.arcgis.com/calcite-design-system/components/). Refer to the [React example](https://github.com/Esri/calcite-design-system/tree/dev/examples/components/react) for a minimal application using this package.

## Deprecation Notice

> [!WARNING]

> > Calcite Components React is deprecated in v5.0.0 and will be fully removed in v6.0.0 (targeted for February 2027).

Reviewer 2: For more direct messaging, should we remove "fully"? Also, we don't generally publicize dates for releases in case timelines shift. Can we remove the date?
```suggestion
> Calcite Components React is deprecated in v5.0.0 and will be removed in v6.0.0.
```

The `@esri/calcite-components-react` package was originally developed for use with React 18 where wrappers were necessary to use custom elements. With React 19's custom element support, these wrappers are no longer needed.

> Consider upgrading to React 19+ and using <code>@esri/calcite-components</code> directly.

Reviewer 2: 🎉

## Why not just use the web components directly?

With React 19+'s native support for custom elements, you can now use <code>@esri/calcite-components</code> directly. See the [Calcite Components React Deprecation Notice](#deprecation-notice) for additional information.

> With React 18, because it uses a synthetic event system, the custom events emitted from Calcite components won't work with JSX in React. For example, say you want to update some value when the `calcite-slider` component changes. When using the standard web components, you need to save a ref to the element, and add a listener:

Reviewer: Slight change to explicitly mention earlier versions. My wording suggestion can probably be improved.

```suggestion
Because React 18 and earlier use a synthetic event system, the custom events emitted from Calcite components won't work with JSX in React. For example, say you want to update some value when the `calcite-slider` component changes. When using the standard web components, you need to save a ref to the element, and add a listener:
```

## Why not just use the web components directly?

With React 19+'s native support for custom elements, you can now use <code>@esri/calcite-components</code> directly. See the [Calcite Components React Deprecation Notice](#deprecation-notice) for additional information.

> Because React 18 and earlier use a synthetic event system, the custom events emitted from Calcite components won't work with JSX in React. For example, say you want to update some value when the `calcite-slider` component changes. When using the standard web components, you need to save a ref to the element, and add a listener:

Reviewer 2: Readability suggestion and small change to a `"Calcite Components"` naming convention we missed in the past: 
```suggestion
Because React 18 and earlier versions use a synthetic event system, the custom events emitted from Calcite Components won't work with JSX in React. For example, if you want to update a value when the `calcite-slider` component changes. When using the standard web components, you need to save a ref to the element, and add a listener:
```

Author: I made some additional edits to this part because the sentence below felt incomplete as is. 

`For example, if you want to update a value when the `calcite-slider` component changes.`
